GetMessage/handler.ts

```diff
@@ -164,6 +170,78 @@ export const getReadStatusForService = (
     O.getOrElse(() => ReadStatusEnum.UNREAD)
   );
 
+const mapPaymentStatus = ({
+  isPaid
+}: PaymentStatusFromUpdater): PaymentStatus =>
+  isPaid ? PaymentStatusEnum.PAID : PaymentStatusEnum.NOT_PAID;
+
+const WithPayment = t.interface({
+  message: t.interface({
+    content: t.interface({ payment_data: PaymentDataWithRequiredPayee })
+  })
+});
+const WithStatusProcessed = t.interface({
+  status: t.literal(MessageStatusValueEnum.PROCESSED)
+});
+const eligibleForPaymentStatus = (
+  messageWithContent: unknown
+): E.Either<t.Errors, unknown> =>
+  pipe(
+    messageWithContent,
+    E.right,
+    E.chainFirst(mwc => WithPayment.decode(mwc)),
+    E.chainFirst(mwc => WithStatusProcessed.decode(mwc))
+  );
```

Comment:
Can this code can be refactored as follow? 
```ts
pipe(
  WithPayment.decode(messageWithContent),
  E.chainFirst(mwc => WithStatusProcessed.decode(mwc))
);
```

Reply:
You could, but in that way the output tipo wille be a `WithPayment` instead of `typeof messageWithContent`.
Using chainFirst for both the decode we only impact the output of left path.